packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -260,16 +270,35 @@ if (__DEV__) {
 }
 
 function ensureListeningTo(
-  rootContainerElement: Element | Node,
+  rootContainerInstance: Element | Node,
   registrationName: string,
 ): void {
-  const isDocumentOrFragment =
-    rootContainerElement.nodeType === DOCUMENT_NODE ||
-    rootContainerElement.nodeType === DOCUMENT_FRAGMENT_NODE;
-  const doc = isDocumentOrFragment
-    ? rootContainerElement
-    : rootContainerElement.ownerDocument;
-  legacyListenToEvent(registrationName, doc);
+  if (enableModernEventSystem) {
+    // If we have a comment node, then use the parent node,
+    // which should be an element.
+    const rootContainerElement =
+      rootContainerInstance.nodeType === COMMENT_NODE
+        ? rootContainerInstance.parentNode
+        : rootContainerInstance;
+    // Containers can only ever be element nodes. We do not
+    // want to register events to document fragments or documents
+    // with the modern plugin event system.
+    invariant(
+      rootContainerElement && rootContainerElement.nodeType === ELEMENT_NODE,
```

Comment:
```suggestion
      rootContainerElement != null && rootContainerElement.nodeType === ELEMENT_NODE,
```
is this 'faster'? or would that be an unnecessary optimisation?